src/pycountry/tests/test_general.py

```diff
@@ -246,3 +284,188 @@ def test_has_version_attribute():
     assert pycountry.__version__ != "n/a"
     assert len(pycountry.__version__) >= 5
     assert "." in pycountry.__version__
+
+
+def test_is_instance_of_language():
+    assert isinstance(pycountry.languages, pycountry.Languages)
+
+
+def test_is_instance_of_country():
+    united_states = pycountry.countries.get(alpha_2="US")
+    class_name = united_states.__class__.__name__
+    assert class_name == "Country"
+
+
+def test_is_instance_of_subdivision():
+    assert isinstance(pycountry.subdivisions, pycountry.Subdivisions)
+
+
+def test_is_instance_of_script():
+    assert isinstance(pycountry.scripts, pycountry.Scripts)
+
+
+def test_is_instance_of_currency():
+    assert isinstance(pycountry.currencies, pycountry.Currencies)
+
+
+def test_add_entry():
+    pycountry.countries._clear()
+    assert pycountry.countries.get(alpha_2="XK") is None
+
+    pycountry.countries.add_entry(
+        alpha_2="XK", alpha_3="XXK", name="Kosovo", numeric="926"
+    )
+
+    country = pycountry.countries.get(alpha_2="XK")
+    assert isinstance(country, pycountry.countries.data_class)
+
+
+def test_remove_entry():
+    pycountry.countries._clear()
+    assert pycountry.countries.get(alpha_2="DE") is not None
+
+    pycountry.countries.remove_entry(alpha_2="DE")
+
+    assert pycountry.countries.get(alpha_2="DE") is None
+
+
+def test_no_results_lookup_error():
+    try:
+        import importlib_resources  # type: ignore
+    except ModuleNotFoundError:
+        from importlib import resources as importlib_resources
+
+    def resource_filename(package_or_requirement, resource_name):
+        return str(
+            importlib_resources.files(package_or_requirement) / resource_name
+        )
+
+    DATABASE_DIR = resource_filename("pycountry", "databases")
+    countries = pycountry.ExistingCountries(
+        os.path.join(DATABASE_DIR, "iso3166-1.json")
+    )
```

Comment:
Is there a reason this test is using a new `ExistingCountries` instance instead of the `pycountry.countries` one?